torch/csrc/distributed/c10d/init.cpp

```diff
@@ -992,6 +1010,36 @@ that adds a prefix to each key inserted to the store.
           "get_future",
           [](::c10d::ProcessGroup::Work& work)
               -> std::shared_ptr<jit::PythonFutureWrapper> {
+#ifdef USE_C10D_NCCL
```

Comment:
Might just be my confusion on this compiler guard. But isn't this true if we compile with NCCL enabled, but the user could still choose the Gloo backend at runtime? What sort of error message do we get if this codepath is executed with the Gloo backend? Could we add a test to ensure something reasonable is raised in this case (can be in a separate diff).

Reply:
There shouldn't be any side effect on other backend, as no code will be executed.